### PR TITLE
[US-256988] Added model to the project. Scene amendments - lights and tap gesture handling

### PR DESCRIPTION
**Pull request by ghawliczek**

### CODE REVIEW

File: iOS/FVM/FVM/FVMCarModelViewController.swift [hunk 1]
```diff
@@ -18,61 +18,42 @@ import SceneKit
...
-            y += 2 * Float(radius)
-        }
-        positionCameraAccordingly(xAmount: xAmount, yAmount: yAmount, radius: radius)
+        
+        let topLight = SCNNode()
+        topLight.light = SCNLight()
+        topLight.light?.type = .directional
+        topLight.position = SCNVector3(x: 0, y: 50, z: 0)
+        topLight.eulerAngles = SCNVector3Make(Float(-Double.pi / 2), 0, 0)
+        scnScene.rootNode.addChildNode(topLight)
```
Comment (SzymonSmykala) [inline iOS/FVM/FVM/FVMCarModelViewController.swift]: Create topLight in method

File: iOS/FVM/FVM/FVMCarModelViewController.swift [hunk 2]
```diff
@@ -18,61 +18,42 @@ import SceneKit
...
+        
+        let bottomLight = SCNNode()
+        bottomLight.light = SCNLight()
+        bottomLight.light?.type = .directional
+        bottomLight.light?.intensity = 750
+        bottomLight.position = SCNVector3(x: 0, y: -50, z: 0)
+        bottomLight.eulerAngles = SCNVector3Make(Float(-Double.pi / 2), 0, 0)
+        scnScene.rootNode.addChildNode(bottomLight)
```
Comment (SzymonSmykala) [inline iOS/FVM/FVM/FVMCarModelViewController.swift]: also create it in method

File: iOS/FVM/FVM/FVMCarModelViewController.swift [hunk 3]
```diff
@@ -81,17 +62,19 @@ open class FVMCarModelViewController : SCNView {
...
+            let highlightedNodes = highlightedParts.map { $0.0 }
+            if highlightedNodes.contains(result.node) {
                 SCNTransaction.begin()
                 SCNTransaction.animationDuration = 0.5
-                highlightedParts.remove(at: highlightedParts.firstIndex(of: result.node)!)
-                result.node.scale = SCNVector3(x: 1, y: 1, z: 1)
+                let hitPartIndex = highlightedNodes.firstIndex(of: result.node)
+                result.node.geometry?.firstMaterial?.diffuse.contents = highlightedParts[hitPartIndex!].material
+                highlightedParts.remove(at: highlightedNodes.firstIndex(of: result.node)!)
                 SCNTransaction.commit()
```
Comment (SzymonSmykala) [inline iOS/FVM/FVM/FVMCarModelViewController.swift]: Code duplication:
`SCNTransaction.begin()
                 SCNTransaction.animationDuration = 0.5		                 SCNTransaction.animationDuration = 0.5`

...
`SCNTransaction.commit()`

Comment (SzymonSmykala) [reply iOS/FVM/FVM/FVMCarModelViewController.swift]: It's seems that Controller class does everything. We have to think how to move some responsibilities to another class.

Comment (ghawliczek) [reply iOS/FVM/FVM/FVMCarModelViewController.swift]: I'd create US for this. For now, leaving as it is.

File: iOS/FVM/FVM/FVMCarModelViewController.swift (hunk 1 above)
Comment (ghawliczek) [reply iOS/FVM/FVM/FVMCarModelViewController.swift]: done

File: iOS/FVM/FVM/FVMCarModelViewController.swift (hunk 2 above)
Comment (ghawliczek) [reply iOS/FVM/FVM/FVMCarModelViewController.swift]: done

**Pull request merged**